### PR TITLE
Add more util docs

### PR DESCRIPTION
**Pull request by mezzode**
Following on from #3931.

### CODE REVIEW

File: keras/utils/data_utils.py [hunk 1]
```diff
@@ -110,6 +124,15 @@ def dl_progress(count, block_size, total_size):
 
 
 def validate_file(fpath, md5_hash):
```
Comment (fchollet) [inline keras/utils/data_utils.py]: This is a single-usage util function and should be left out (e.g. added to `EXCLUDE` in `autogen.py`.

File: keras/utils/layer_utils.py [hunk 2]
```diff
@@ -37,8 +37,14 @@ def layer_from_config(config, custom_objects={}):
 
 
 def print_summary(layers, relevant_nodes=None, line_length=100, positions=[.33, .55, .67, 1.]):
```
Comment (fchollet) [inline keras/utils/layer_utils.py]: Same, this one should be left out.

File: docs/autogen.py [hunk 3]
```diff
@@ -244,11 +247,23 @@
...
             io_utils.HDF5Matrix
         ],
     },
+    {
+        'page': 'utils/layer_utils.md',
+        'all_module_functions': [layer_utils]
+    },
+    {
+        'page': 'utils/np_utils.md',
+        'all_module_functions': [np_utils]
```
Comment (fchollet) [inline docs/autogen.py]: These new pages should also be added in `mkdocs.yml`.

**Comment by mezzode** [comment]
Removed the single use utils, and the new pages have already been added to `mkdocs.yml`

**Comment by fchollet** [comment]
LGTM

**Pull request merged**